src/Utility/dontYield.lua

```diff
@@ -0,0 +1,52 @@
+--!nonstrict
+
+--[[
+    Utility function to throw an error when a function yields
+
+    You can either provide a specific error message using the second
+    parameter (errorMessage: string?), or you can let it default to the
+    default error message (DEFAULT_ERROR_MESSAGE). Error message here
+    means additional information that the user might find helpful to
+    determine where the yielding is occurring. (i.e. "Cannot yield inside
+    of the ComputedPairs processor function.",)
+]]
+
+local Package = script.Parent.Parent
+local logError = require(Package.Logging.logError)
+
+local DEFAULT_ERROR_MESSAGE: string = "N/A"
+
+--[[
+    Handles the return values resulting from calling `coroutine.resume(func, ...)`
+    Makes it so we don't have to use table.unpack to get the Variant returned from
+    calling `coroutine.resume(func, ...)`
+]]
+local function handleResult(taskCoroutine, errorMessage, success, ...)
+    if success then
+        if coroutine.status(taskCoroutine) == "dead" then
+            -- we didn't yield
+            return ...
+            
+        else
+            -- we yielded
+            logError("cannotYield", ({...})[1], errorMessage or DEFAULT_ERROR_MESSAGE)
```

Comment:
It's better DX to allow calling code to provide a custom error ID, rather than using a generic message and formatting it with custom information.

This allows for more contextually relevant error messages, as well as an improved search experience thanks to using unique error IDs for unique situations:

> Can't yield inside a Computed (ID: computedCannotYield)

as opposed to:

> Attempted to yield in a synchronous thread. Information: Cannot yield inside of the Computed callback function. (ID: cannotYield)

Reply:
I experimented with doing this initially, but for some reason I ended up not doing it... can't remember why since I am definitely in agreement with you on this one. I'll work on addressing that.

Reply:
I pushed a fix for this, but I am leaving the comment unresolved so you can verify this fix properly address your comment! :)